docker-charming-planning.md

```diff
@@ -0,0 +1,74 @@
+# Charming with Docker
+
+> This is a guideline reference for planning the docker charming tutorial
+> and makes some assumptions about the introduction of the technology, plus
+> stack of tech that will be used to create these charms.
+>
+> Any user coming into this ecosystem is assumed to understand how to create
+> docker images, but have no understanding of juju charming.
+
+
+## A narrative view into consuming Docker with Juju Charms
+
+
```

Comment:
I'm thinking there are 2 kinds of developers: those who want to get something working and those who need to understand how something works.   Usually this is a temporal situation, "I made it work, but now I need to understand it"

To that end, I think we should make this bimodal, sort of a dive into docker charming that can break out into the sections you have here.   Idea is that folks can get something working, then explore it.